go/enclave/rollupchain/rollup_chain.go

```diff
@@ -894,27 +890,56 @@ func (rc *RollupChain) isAccountContractAtBlock(accountAddr gethcommon.Address,
 }
 
 // Validates and stores the rollup in a given block.
-func (rc *RollupChain) processRollups(rollups []*core.Rollup, _ *gethcommon.Hash) error {
-	if len(rollups) == 0 {
-		// TODO - #718 - Store previous rollup against new L1 head.
-		return nil
+func (rc *RollupChain) processRollups(rollups []*core.Rollup, block *common.L1Block) error {
+	// We sort the rollups by number in ascending order, in order to process them in the correct order.
+	sort.Slice(rollups, func(i, j int) bool {
+		return rollups[i].Header.Number.Cmp(rollups[j].Header.Number) < 0
+	})
+
+	// We check if there are any duplicates.
+	for idx, rollup := range rollups {
+		if idx+1 >= len(rollups) {
+			break
+		}
+		if rollup.Header.Number.Cmp(rollups[idx+1].Header.Number) == 0 {
+			return fmt.Errorf("duplicates rollups found in block; two rollups with number %d", rollup.Header.Number)
+		}
 	}
 
+	// We retrieve the current head rollup.
+	l1ParentHash := block.ParentHash()
+	currentHeadRollup, err := rc.storage.FetchHeadRollupForBlock(&l1ParentHash)
+	if err != nil {
+		return fmt.Errorf("could not fetch current L2 head rollup")
+	}
+
+	// We check each rollup.
 	for _, rollup := range rollups {
-		if err := rc.validateSequencerSig(rollup.Header.Hash(), rollup.Header.Agg, rollup.Header.R, rollup.Header.S); err != nil {
+		if err := rc.validateSequencerSig(rollup.Hash(), &rollup.Header.Agg, rollup.Header.R, rollup.Header.S); err != nil {
 			return fmt.Errorf("rollup signature was invalid. Cause: %w", err)
 		}
 
-		// TODO - #718 - Also validate the rollups in the block against the stored batches.
-		//  * Rollups link to previous rollup
-		//  * Rollups contains all expected batches
+		// We check that the rollup numbers are sequential.
+		rollupNumIncrement := big.NewInt(0).Sub(rollup.Number(), currentHeadRollup.Number())
```

Comment:
Maybe better to check parent hashes rather than rollup number? Or at least leave a todo for that I guess.